### PR TITLE
[Replicate] compatibility with PHP 5.x

### PR DESCRIPTION
**Pull request by TingDaoK**
*Issue #, if available:*

#95 
original: https://github.com/awslabs/aws-crt-php/pull/96

*Description of changes:*

- Make sure run the test... Run the test for php5.5
- Add readme section about the mismatch of libcrypto from php and awscrt. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -37,13 +37,8 @@ jobs:
...
         ./configure
-        make
-
-    - name: Install dependencies
-      # get path to composer.phar so we can run it with custom php.ini
-      run: |
-        COMPOSER_PHAR=$(realpath $(which composer))
-        php -c php.ini $COMPOSER_PHAR update --no-interaction
+        USE_OPENSSL=OFF make
+        ./dev-scripts/run_tests.sh
```
Comment (waahm7) [inline .github/workflows/ci.yml:45]: Debatable: maybe we should update our CI to run it with the same version of OpenSSL instead of AWS-LC.

**Review by waahm7 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (graebm) [inline .github/workflows/ci.yml:45]: add comments explaining hack
```suggestion
        # Normally, CRT uses the system's default libcrypto.so.
        # But in this CI run, PHP 5.5 is using a different older libcrypto.so.
        # When an applications uses two different libcrypto.so at the same time, things tend to explode.
        # Therefore, in this CI run we build AWS-LC statically, and use its libcrypto.a instead.
        USE_OPENSSL=OFF make
        ./dev-scripts/run_tests.sh
```

**Review by graebm — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (graebm) [reply .github/workflows/ci.yml:45]: we tried, but it was going to take a lot more work, and this seemed like a reasonable workaround for now

Comment (TingDaoK) [reply .github/workflows/ci.yml:45]: added a bit more to explain that php5.5 cannot build with openssl>=1.1 referring to https://github.com/shivammathur/setup-php/issues/714#issuecomment-1480427721

**Pull request merged**